From d2d23fb6089657846f6b96ea2a8cc3b4603020ea Mon Sep 17 00:00:00 2001
From: Mike McQuaid <mike@mikemcquaid.com>
Date: Thu, 12 Aug 2010 13:58:01 +0100
Subject: [PATCH] Add support for making bundle items writable before fixing up.

This ensure that any bundle items are made user writable before
any attempt is made to alter them using install_name_tool. This is
because MacPorts/Fink/Homebrew don't install libraries as writable
so this fix is needed to allow fixup_bundle_item to work correctly
when using these package managers.
---
 Modules/BundleUtilities.cmake |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/Modules/BundleUtilities.cmake b/Modules/BundleUtilities.cmake
index 7690094..82f94b9 100644
--- a/Modules/BundleUtilities.cmake
+++ b/Modules/BundleUtilities.cmake
@@ -499,6 +499,9 @@ endfunction(copy_resolved_framework_into_bundle)
 # Accumulate changes in a local variable and make *one* call to install_name_tool
 # at the end of the function with all the changes at once.
 #
+# If the chmod_bundle_items variable is set then any bundle items will be set as
+# writable before install_name_tool tries to change them.
+#
 function(fixup_bundle_item resolved_embedded_item exepath dirs)
   # This item's key is "ikey":
   #
@@ -521,6 +524,10 @@ function(fixup_bundle_item resolved_embedded_item exepath dirs)
     endif(NOT "${${rkey}_EMBEDDED_ITEM}" STREQUAL "")
   endforeach(pr)
 
+  if (chmod_bundle_items)
+    execute_process(COMMAND chmod u+w "${resolved_embedded_item}")
+  endif (chmod_bundle_items)
+
   # Change this item's id and all of its references in one call
   # to install_name_tool:
   #
-- 
1.7.2.1

